core/domain/stats_jobs_one_off.py

```diff
@@ -26,14 +32,246 @@
+                    state_stats.num_times_solution_viewed_v2 = 0
+                    state_stats.num_completions_v2 = 0
+                exp_stats_dict = RecomputeStatistics._recompute_statistics(
+                    exp_id, version, events, v1_stats.to_dict())
+                exp_stats_dicts.append(exp_stats_dict)
+            else:
+                change_list = exp_models.ExplorationCommitLogEntryModel.get(
+                    'exploration-%s-%s' % (exp_id, version)).commit_cmds
+
+                # Copy recomputed v2 events from previous version
+                prev_version = copy.deepcopy(exp_stats_dicts[-1])
+
+                # Copy uncorrupt v1 stats
+                prev_version['num_starts_v1'] = v1_stats.num_starts_v1
+                prev_version['num_completions_v1'] = v1_stats.num_completions_v1
+                prev_version['num_actual_starts_v1'] = (
+                    v1_stats.num_actual_starts_v1)
+                state_stats_mapping = prev_version['state_stats_mapping']
+                for state in state_stats_mapping.keys():
+                    state_stats_mapping[state]['total_answers_count_v1'] = (
+                        v1_stats.state_stats_mapping[state]
+                        .total_answers_count_v1)
+                    state_stats_mapping[state]['useful_feedback_count_v1'] = (
+                        v1_stats.state_stats_mapping[state]
+                        .useful_feedback_count_v1)
+                    state_stats_mapping[state]['total_hit_count_v1'] = (
+                        v1_stats.state_stats_mapping[state]
+                        .total_hit_count_v1)
+                    state_stats_mapping[state]['first_hit_count_v1'] = (
+                        v1_stats.state_stats_mapping[state]
+                        .first_hit_count_v1)
+                    state_stats_mapping[state]['num_completions_v1'] = (
+                        v1_stats.state_stats_mapping[state]
+                        .num_completions_v1)
+
+                exploration_stats = (
+                    RecomputeStatistics._update_state_stats_mapping(
+                        prev_version, change_list))
+
+                exp_stats_dict = RecomputeStatistics._recompute_statistics(
+                    exp_id, version, events, exploration_stats)
+                exp_stats_dicts.append(exp_stats_dict)
+
+        stats_models.ExplorationStatsModel.save_multi(exp_stats_dicts)
+
+    @classmethod
+    def _update_state_stats_mapping(cls, old_stats_model, change_list):
+        """Update the state_stats_mapping to correspond with the changes
+        in change_list.
+
+        Args:
+            old_stats_model: dict. A dict representation of an
+                ExplorationStatsModel.
+            change_list: list(dict). A list of all of the commit cmds to
+                the old_stats_model up to the next version.
+
+        Returns:
+            dict. An dict representation of an ExplorationStatsModel
```

Comment:
nit: A dict representation*

Reply:
@LanJosh -- this nit still applies, please check previous comments and resolve them by replying "Done." after you've made the fix.

Reply:
Done.